Zend/zend_compile.c

```diff
@@ -5671,8 +5671,8 @@ static zend_type zend_compile_typename(
 			ZSTR_VAL(type_str));
 	}
 
-	if ((type_mask == MAY_BE_ANY) && allow_null) {
-		zend_error_noreturn(E_COMPILE_ERROR, "Type mixed cannot be nullable");
+	if (type_mask == MAY_BE_ANY && allow_null && !force_allow_null) {
```

Comment:
I think this is still not quite right: What about `?mixed $arg = null`?

Reply:
:( I really hope that I got it right this time...